github_activity/github_activity.py

```diff
@@ -486,12 +486,19 @@ def _get_datetime_and_type(org, repo, datetime_or_git_ref):
         dt = datetime.datetime.now().astimezone(pytz.utc)
         return (dt, False)
 
-    if _valid_git_reference_check(datetime_or_git_ref):
+    try:
         dt = _get_datetime_from_git_ref(org, repo, datetime_or_git_ref)
         return (dt, True)
-    else:
-        dt = dateutil.parser.parse(datetime_or_git_ref)
-        return (dt, False)
+    except Exception as ref_error:
+        try:
+            dt = dateutil.parser.parse(datetime_or_git_ref)
+            return (dt, False)
+        except Exception as datetime_error:
+            raise ValueError(
+                "{0} not found as a ref or valid date format".format(
+                    datetime_or_git_ref
+                )
+            )
```

Comment:
As part of this refactoring, _valid_git_reference_check is no longer used. Can you delete its definition and the import of it as well?

Reply:
It is very helpful that you write `except Exception as datetime_error:` even though datetime_error isn't used for readability.